api/services/control/errors.go

```diff
@@ -0,0 +1,74 @@
+package moby_buildkit_v1
+
+import (
+	context "context"
+	"log"
+
+	"github.com/moby/buildkit/solver/errdefs"
+	grpc "google.golang.org/grpc"
+)
+
+func WrapServerErrors(srv ControlServer) ControlServer {
```

Comment:
maybe I should do `grpc.StreamInterceptor` `grpc.UnaryInterceptor` instead. Not much cleaner and need to merge with existing ones. But no changes required on new API endpoints.